python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -438,6 +441,9 @@ def flush(self):
                 subjob_data = self.__get_dataFile(str(index))
                 subjob_obj = self._cachedJobs[index]
 
+                if subjob_obj is subjob_obj._getRoot():
+                    raise GangaException(self, "Subjob parent not set correctly in flush.")
```

Comment:
I see the point but I'd be wary of putting more locks in at present, at least until we see an issue. Also, I would expect the worse that can happen is that two threads load the same job at slightly different times and the second overwrites the first but with the same info. In which case, I don't think it would matter.